src/main/java/mods/railcraft/common/plugins/dynamiclights/DynamicLightsPlugin.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) CovertJaguar, 2014 http://railcraft.info
+ *
+ * This code is the property of CovertJaguar
+ * and may only be used with explicit written
+ * permission unless otherwise specified on the
+ * license page at http://railcraft.info/wiki/info:license.
+ */
+package mods.railcraft.common.plugins.dynamiclights;
+
+import atomicstryker.dynamiclights.client.IDynamicLightSource;
+import mods.railcraft.common.util.misc.Game;
+import net.minecraft.entity.Entity;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.event.entity.EntityJoinWorldEvent;
+import net.minecraftforge.fml.common.FMLCommonHandler;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import org.apache.logging.log4j.Level;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.ToIntFunction;
+
+/**
+ * A plugin for compatibility with
+ * <url https://github.com/AtomicStryker/atomicstrykers-minecraft-mods/blob/1.10/DynamicLights/>
+ * Dynamic Lights</url>
+ */
+public class DynamicLightsPlugin {
+    public static final String MOD_ID = "DynamicLights";
+    private static final DynamicLightsPlugin INSTANCE = new DynamicLightsPlugin();
+
+    private Handle handle;
+
+    public static DynamicLightsPlugin getInstance() {
+        return INSTANCE;
+    }
+
+    public void registerEntityLightSource(Class<? extends Entity> type, ToIntFunction<Entity> lightCalculator) {
+        handle.registerEntityLightSource(type, lightCalculator);
+    }
+
+    private DynamicLightsPlugin() {
+        if (FMLCommonHandler.instance().getSide().isClient()) { //only enable on client
+            try {
+                Class<?> dynamicLightsClass = Class.forName("atomicstryker.dynamiclights.client.DynamicLights");
+                Class<?> iDynamicLightSourceClass = IDynamicLightSource.class;
```

Comment:
Why is there a hard link for IDynamicLightSource but a soft link for DynamicLights?